packages/inventory-compliance/src/ComplianceRemediationButton.js

```diff
@@ -53,9 +57,17 @@ class ComplianceRemediationButton extends React.Component {
         });
     }
 
-    noRemediationAvailable = () => {
+    remediationAvailable = () => {
         const { allSystems, selectedRules } = this.props;
-        return selectedRules.length === 0 && !allSystems.some((system) => system.ruleObjectsFailed.some((rule) => rule.remediationAvailable && system.supported));
+        let rules = selectedRules.length ? selectedRules : allSystems.flatMap((system) => system.ruleObjectsFailed);
+        return rules.some((rule) => (
+            rule.remediationAvailable &&
+            (
+                rule.profiles?.some((profile) => profile.supported) ||
+                allSystems.some((system) => this.ruleProfile(rule, system).supported)
+            ) &&
+            rule.compliant === false
```

Comment:
Nitpick: I'd recommend to move simpler condition at the top. I know that we do not program on Intel 8080 and this is a premature optimization. If we'd have lots of systems and rules, it could make an impact.